### ISSUE TITLE
Amazon S3 Replication to multiple destinations

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/terraform-provider-aws#16546]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

> Amazon S3 Replication now gives you the ability to replicate data from one source bucket to multiple destination buckets in the same, or different AWS Regions.

### New or Affected Resource(s)

* [`aws_s3_bucket`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket)

### Potential Terraform Configuration

```hcl
resource "aws_s3_bucket" "source" {
  bucket = "example"

  versioning {
    enabled = true
  }

  replication_configuration {
    role = aws_iam_role.replication.arn

    rules {
      id       = "rule1"
      priority = 1
      status   = "Enabled"

      filter {}

      destination {
        bucket        = aws_s3_bucket.destination1.arn
        storage_class = "STANDARD"
      }
    }

    rules {
      id       = "rule2"
      priority = 2
      status   = "Enabled"

      filter {
        prefix = "prefix2"
      }

      destination {
        bucket        = aws_s3_bucket.destination2.arn
        storage_class = "STANDARD_IA"
      }
    }
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/12/amazon-s3-replication-adds-support-for-multiple-destinations-in-the-same-or-different-aws-regions/).
[Blog post](https://aws.amazon.com/blogs/aws/new-amazon-s3-replication-adds-support-for-multiple-destination-buckets/).
[Developer Guide](https://docs.aws.amazon.com/AmazonS3/latest/dev/replication-add-config.html#replication-config-optional-dest-config).

**Comment by ewbankkit** [comment]
It looks like this _should_ be supported in the current code (there is no limitation on `rules` that would prevent this scenario).
We should add an explicit acceptance tests case for this.

**Comment by maratdr** [comment]
> It looks like this _should_ be supported in the current code (there is no limitation on `rules` that would prevent this scenario).
> We should add an explicit acceptance tests case for this.

It seems to be limited by the sdk version of terraform aws provider (3.20.0). At moment, I am still getting below error
```
Error: Error putting S3 replication configuration: InvalidRequest: Number of distinct destination bucket ARNs cannot exceed 1
	status code: 400, request id: C0A5C4E04398B311, host id: vEVcHcwOS2lN6DRX9Twqt8gwBcF8tf9DM94PGKgtA0op6JmfgPvvBaA3Z8z1bL5oLOCOvnFcE8M=
```

**Comment by ewbankkit** [comment]
Hmm, **v3.20.0** built against AWS SDK v1.36.0: https://github.com/hashicorp/terraform-provider-aws/pull/16530.
I'll add a new acceptance test and try with the latest trunk code.

**Comment by ewbankkit** [comment]
Priority is used when multiple destinations are specified, so `priority` must be specified.
Currently we are only sending `priority` if a `filter` is also specified -that needs to be addressed for the example above to work.

**Comment by geototti21** [comment]
👋 hey @ewbankkit, I come about on this error `Number of distinct destination bucket ARNs cannot exceed 1` , and while I set priority to all rules I kept getting this error even on latest version of the provider (3.21).

The problem was that I was not specifying `filter {}` as I don't have any filters at all. The moment I specified this it worked, even as an empty object. Not sure what will be the best behavior here but just in case anyone else has the same problem. 

Maybe we can update the docs?
